index.js

```diff
@@ -1,19 +1,37 @@
 'use strict';
-module.exports = (condition, interval) => new Promise((resolve, reject) => {
+module.exports = (condition, interval, timeout) => new Promise((resolve, reject) => {
 	interval = typeof interval === 'number' ? interval : 20;
 
+	let timedOut = false;
+	let timer;
+
+	if (typeof timeout === 'number') {
```

Comment:
This would have been a lot cleaner if `setTimeout(() => {}, Infinity)` didn't just run immediately :(